roles/openshift_hosted/tasks/registry/storage/s3.yml

```diff
@@ -36,13 +36,16 @@
       - path: cloudfront.pem
         data: "{{ lookup('file', openshift_hosted_registry_storage_s3_cloudfront_privatekeyfile) }}"
 
-  - name: Add cloudfront secret to the registry volumes
+  - name: Create cloudfront secret registry volume
     set_fact:
       s3_volume_mount:
       - name: cloudfront-vol
         path: /etc/origin
         type: secret
         secret_name: docker-registry-s3-cloudfront
+
+  - name: Append cloudfront secret registry volume to openshift_hosted_registry_volumes
+    set_fact:
       openshift_hosted_registry_volumes: "{{ openshift_hosted_registry_volumes | union(s3_volume_mount) }}"
```

Comment:
I think there has to be a better way than using set_fact, but that would require quite a bit of refactoring.

This should be able to be brought down to a single set_fact by doing something like this:
```
- name: Append cloudfront secret registry volume to openshift_hosted_registry_volumes
  set_fact:
    openshift_hosted_registry_volumes: "{{ openshift_hosted_registry_volumes | union(s3_volume_mount) }}"
  vars:
    s3_volume_mount:
    - name: cloudfront-vol
      path: /etc/origin
      type: secret
      secret_name: docker-registry-s3-cloudfront
```